src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int DEFAULT_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[DEFAULT_SIZE];
+        this.values = (V[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        Integer index = getIndex(key);
+        if (index != null) {
+            values[index] = value;
+        } else {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        Integer index = getIndex(key);
+        return index != null ? (V) values[index] : null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private Integer getIndex(K key) {
+        Integer result = null;
+        for (int i = 0; i < size; i++) {
+            if ((keys[i] != null && keys[i].equals(key))
+                    || (keys[i] == key && keys[i] == null)) {
+                result = i;
+            }
+        }
+        return result;
```

Comment:
```suggestion
    private int getIndex(K key) {
        for (int i = 0; i < size; i++) {
            if (keys[i] == key
                    || (keys[i] != null && keys[i].equals(key)) {
                return i;
            }
        }
        return -1;
```
index of an element in an array is usually an int value